Source/DafnyCore/Verifier/Translator.TrStatement.cs

```diff
@@ -505,7 +505,6 @@ public partial class Translator {
         // try/recover statements are currently internal-only AST nodes that cannot be
         // directly used in user Dafny code. They are only generated by rewriters, and verifying
         // their use is low value.
-        throw new NotSupportedException("Verification of try/recover statements is not supported");
```

Comment:
`RunAllTestsMainMethod` runs post resolution, before verification, and no longer changes just the CompileModules, so the try/recover statements end up in verified code.

Running that transformation after verification causes slight changes in warning/error behavior, since then diagnostics from `RunAllTestsMainMethod` are reported after resolution.

Reply:
It seems to me that it'd be better to run it after verification, even if it changes warning/error behavior, since here we're adding some code to what will be verified and then just skipping verification of it. That seems fragile to me, and like it could potentially introduce soundness issues. Am I missing something?

Reply:
Previously the code added by `RunAllTestsMainMethod` was entirely unverified, and after this PR it is verified except for the try/recover statement since we didn't implement verification for it. Doesn't that make it more instead of less sound?